src/Nethermind/Nethermind.Evm/EvmState.cs

```diff
@@ -127,6 +127,7 @@ private void Initialize(
         ReturnStackHead = 0;
         ExecutionType = executionType;
         ProgramCounter = 0;
+        FunctionIndex = 0;
```

Comment:
This was the cause of some obscure fuzzing breakage.  If a transaction exited (such as Out of Gas) will in a CALLF then the old function index would be used at the start of the next transaction, resulting in line 1 having an incorrect section number.

Reply:
And to clarify, this was a test instrumentation issue only, it did no impact consensus.